### PR TITLE
[inductor] Allow `tensors` kwarg in sink_cat_after_pointwise

### PR DESCRIPTION
**Pull request by bertmaher**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #97028
* __->__ #97019

Lacking handling of kwargs strikes again.

Differential Revision: [D44166740](https://our.internmc.facebook.com/intern/diff/D44166740/)

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

File: torch/_inductor/overrides.py [hunk 1]
```diff
@@ -299,13 +316,17 @@ def is_pointwise_unary(node):
 
         if user and is_pointwise_unary(user):
             with g.inserting_before(node):
+                norm_cat = NormalizedCatNode(node)
```
Comment (jansel) [inline torch/_inductor/overrides.py:319]:
```suggestion
                def cat_args(tensors, dim):
                   return tensors, dim
                tensors, dim = cat_args(*node.args, **node.kwargs)
```

**Review by jansel — APPROVED** [review]
Yet another reason to do optimizations after AotAutograd (which has normalized args for you already)

File: torch/_inductor/overrides.py (hunk 1 above)
Comment (bertmaher) [reply torch/_inductor/overrides.py:319]: Oh that's cute haha.  I'll take it!

**Comment by bertmaher** [comment]
And yes, I want to move this (and maybe other transforms) after AOTAutograd for that reason.  I'll take a swing at it next week.

**Comment by ngimel** [comment]
But don't you want AOTAutograd to run on a transformed graph so that backward is also benefiting from this?

**Comment by jansel** [comment]
That is correct.  Maybe we would need a pattern for the backwards as well?

**Comment by bertmaher** [comment]
@bertmaher has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D44166740).

**Pull request closed without merging**